### ISSUE TITLE
How do you get the point clouds of second frame for KITTI?

### ISSUE THREAD

**Issue by sulashi** [laoreja/HPLFlowNet#14]
Thank you for your great work and code release!
I found that the disparity of KITTI scene flow dataset is based on frame one, so that the disparity of pixels for the second frame can not be gotten directly. How dou you get them, from the raw lidar data or some method else? 
I guessed you use raw lidar data based on that you say " we evaluate on all 142 scenes in the training set with publicly available raw 3D data, following [25]" in your paper. However, I just found that your PC2 is corresponding with PC1 on every point both from the KITTI scene flow dataset in your code? 

So if you did not use the raw lidar data, why do you use 142 frames rather than 200?

**Comment by laoreja** [comment]
Hi, please refer https://github.com/laoreja/HPLFlowNet/blob/master/data_preprocess/process_kitti.py for data preprocessing. I think i used the ground truth optical flow to warp the disparity for the 2nd frame.

Initially the 2 point clouds are corresponding to each other, by sampling N points from each point cloud, they become not corresponding to each other (but for the full point clouds without sampling, they are).

I used the ones with publicly available raw 3D data for they have less noise. I reported the performance on 200 frames in supp.

Forgive me if I made any mistake, the code is finished 1 year ago. Thanks!

**Comment by sulashi** [comment]
Thank you! You have solved my confusion. 

I wonder if you can selecte all points in 1st frame. Or you sample N points for 1 times for the evaluation results on KITTI?

**Comment by laoreja** [comment]
The sampling is done randomly on the fly and is controlled by a parameter that sets how many points are sampled: `num_points: 8192` in configs.

The two frames are sampled separately, so the points in the 1st frame do not have exact correspondents in the 2nd frame.

If I remember correctly, when setting the number of sampled points to -1, then all the points are used. In this case, there will be no sampling.

During training, I only used 8192 points, otherwise there will be OOM. During testing on KITTI, it's okay to use all the points if your GPU has 12G memory. This approach works well because we have the density normalization.
